v1 draft, NG approval variables CDS-74197 (#3512)
harness/developer-hub: docs/platform/variables-and-expressions/harness-variables.md

* `<+artifacts.sidecars.SIDECAR_IDENTIFIER.tag>`
* `<+artifacts.sidecars.SIDECAR_IDENTIFIER.connectorRef>`

## Approval

> Whenever a user grants an approval in an Approval step, the pipeline maintains the user information of the approver for the rest of the pipeline execution.  You can use these variables in notifications after an approval gets granted.  

Reviewer: ```suggestion
Whenever a user grants an approval in an Approval step, the pipeline maintains the user information of the approver for the rest of the pipeline execution. You can use these variables in notifications after an approval is granted.  
```

These variables are available for Approval steps only, not stages. 

> In the following example, a Deploy stage has two Approval steps. For each approval, the pipeline maintains a separate set of approval variables. Use the index to access the array element that contains the variables for the relevant approval.

Author: ```suggestion
In the following example, a Deploy stage has two Approval steps. For each approval, the pipeline maintains a separate set of approval variables. Use the array index to access the variables for a specific approval.
```

> In the following example, a Deploy stage has two Approval steps. For each approval, the pipeline maintains a separate set of approval variables. Use the index to access the array element that contains the variables for the relevant approval.

Reviewer: How about this:
To access the variables associated with a specific approval, use the index to reference the array element containing those variables.

![](./static/approved-user-pipeline-example.png)

### <+approval.approvalActivities[0].user.name>

The Harness username of the approver.

### <+approval.approvalActivities[0].user.email>

The email address of the approver.

### <+approval.approvalActivities[0].comments>

> Any comments the user added to the approval, formatted as a single string. This variable is populated from the Comment output variable in the approval step or stage.  

Author: ```suggestion
User comments from the approval, formatted as a single string. This variable is populated from the Comment output variable generated by the approval step.  
```

> Any comments the user added to the approval, formatted as a single string. This variable is populated from the Comment output variable in the approval step or stage.  

Reviewer: How about this:
User comments from the approval process are now consolidated into a single formatted string. This string is populated using the 'Comment' output variable from the approval step or stage.
